lib/tokens.js

```diff
@@ -449,6 +469,64 @@ export async function resolveToInternalId({token}) {
   return {internalId};
 }
 
+/**
+ * Sets the `minAssuranceForResolution` on the token batches associated
+ *   `internalId`.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.internalId - The internal ID of the token batch.
+ * @param {number} options.minAssuranceForResolution - Minimum level of
+ *   assurance required for token resolution.
+ *
+ * @returns {Promise<boolean>} Returns true if update occurred.
+ */
+export async function setMinAssuranceForResolution({
+  internalId, minAssuranceForResolution
+}) {
+  const query = {
+    'tokenBatch.internalId': internalId,
```

Comment:
Similar problem here -- we can't do a query on `tokenBatch.internalId` -- we have no such index. We should have a meeting that includes @mattcollier to discuss options here.

Reply:
@mandyvenables, so the decision from the meeting was to add a context for `tokenBatch.internalId` to `tokenization-tokenBatch` -- even though it will not include the shard key. Since we don't expect this code to execute in the hottest code paths, we're going to give the "scatter-gather" approach an attempt and test to see how it affects performance in the future. Please add the above index to this PR. Thanks!

Reply:
I added a separate index for `tokenBatch.internalId` to `tokenization-tokenBatch`.

Reply:
Thanks!